### ISSUE TITLE
Standard way of sharing architectures, e.g., to PyPi.org

### ISSUE THREAD

**Issue by SamuelMarks** [keras-team/keras#15762]
There are a huge number of new statistical, machine-learning and artificial intelligence solutions being released every month.

Most are open-source and written in a popular Python framework like TensorFlow, JAX, or PyTorch.

In order to 'guarantee' you are using the best [for given metric(s)] solution for your dataset, some way of automatically adding these new statistical, machine-learning and artificial intelligence solutions to your automated pipeline needs to be created.

(additionally: useful for testing your new optimiser, loss function, &etc. across a zoo of datasets)

Ditto for transfer learning models. A related problem is automatically putting ensemble networks together. Something like:

```python
import some_broke_arch  # pip install some_broke_arch
import other_neat_arch  # pip install other_neat_arch
import horrible_v_arch  # builtin to keras

model   = some_broke_arch.get_arch(   **standard_arch_params  )
metrics = other_neat_arch.get_metrics(**standard_metric_params)
loss    = horrible_v_arch.get_loss(   **standard_loss_params  )

model.compile(loss=loss, optimizer=keras.optimizers.RMSprop, metrics=metrics)
print(model.summary())
# &etc.
````

In summary, I am petitioning for standard ways of:

  0. exposing algorithms for consumption;
  1. combining algorithms;
  2. comparing algorithms.

To that end, I would recommend encouraging the PyPi folk to add a few new [classifiers](https://pypi.org/classifiers), and a bunch of us trawl through GitHub every month sending PRs to random repositories—associated with academic papers—linking up with CI/CD so that they are now installable with `pip install` and searchable by classifier on PyPi.

Related:

  - my open-source multi-ML meta-framework;
      - uses builtin [`ast`](https://docs.python.org/3/library/ast.html) and [`inspect`](https://docs.python.org/3/library/inspect.html) modules to traverse the module, class, and function hierarchy for 10 popular open-source ML/AI frameworks;
      - will enable experimentation with entire 'search-space' of all these ML frameworks (every transfer learning model, optimiser, loss function, &etc.)
      - with a standard way of sharing architectures will be able to expand the 'search-space' with community contributed solutions.
  - this issue from Jul 20, 2019: https://github.com/tensorflow/tensorflow/issues/30896

**Comment by haifeng-jin** [comment]
Triage notes:

In the short-term, we don't have enough bandwidth for supporting this feature.
but we will keep it open for consideration.

**Comment by SamuelMarks** [comment]
@haifeng-jin I can add this feature and maintain it on my own org. The issue is no one knows who I am or have heard of my org. So sending PRs around the place and requested PyPi add new classifiers is likely to fall on deaf ears.

As for exposing the intricacies of Keras, TensorFlow, JAX, PyTorch, &etc… that's what multi-ML meta-framework—written with my compiler—is for. So I won't need major changes made to any of these frameworks (only minor changes to conform types to expected inputs & outputs).

What I do need is enough reputation to push the community from a hodge-podge collect of adhoc Jupyter Notebooks, scripts, and python packages to conformance to a new specification (PyPi I think is still best… RFC welcome).

**Comment by haifeng-jin** [comment]
@SamuelMarks You are welcome to build your own package to do it, but, unfortunately, we are not able to officially support it right now.

**Comment by bhack** [comment]
@SamuelMarks try to contact https://github.com/mlcommons/ck/blob/master/ck2/docs/motivation.md

They could be interested in something similar.
